src/main/java/net/dv8tion/jda/core/entities/Message.java

```diff
@@ -1254,4 +1262,247 @@ public boolean isImage()
             return height > 0 && width > 0;
         }
     }
+
+    /**
+     * Represents a {@link net.dv8tion.jda.core.entities.Message} activity like Spotify.
+     */
+    class Activity
+    {
+        private final ActivityType type;
+        private final String partyId;
+        private final Application application;
+
+        /**
+         *
+         * @param  type the {@link net.dv8tion.jda.core.entities.Message.ActivityType}
+         *         for the {@link net.dv8tion.jda.core.entities.Message.Activity}.
+         *
+         * @param  partyId the partyId for the {@link net.dv8tion.jda.core.entities.Message.Activity}.
+         *
+         * @param  application the {@link net.dv8tion.jda.core.entities.Message.Application}
+         *         for the {@link net.dv8tion.jda.core.entities.Message.Activity}.
+         *
+         * @throws java.lang.IllegalArgumentException
+         *         <ul>
+         *             <li>If {@link net.dv8tion.jda.core.entities.Message.ActivityType} requires a not null {@link net.dv8tion.jda.core.entities.Message.Application}.</li>
+         *             <li>If {@link net.dv8tion.jda.core.entities.Message.ActivityType} gets a not required not null {@link net.dv8tion.jda.core.entities.Message.Application}.</li>
+         *         </ul>
+         */
+        public Activity(ActivityType type, String partyId, Application application)
+        {
+            this.type = type;
+            this.partyId = partyId;
+            if ((type == ActivityType.GAME && application == null) || (type != ActivityType.GAME && application != null))
+                throw new IllegalArgumentException("Either the ActivityType is wrong or the Application is null!");
```

Comment:
Use the `Checks` helper with Checks.notNull etc

Reply:
I do i was waiting for a response.